modules/bimi.jsm.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Brand Indicators for Message Identification (BIMI).
+ * https://datatracker.ietf.org/doc/draft-brand-indicators-for-message-identification/04/
+ *
+ * BIMI implementation for a Mail User Agent (MUA).
+ * Gets the BIMI Indicator based on the information the receiving
+ * Mail Transfer Agent (MTA) writes into the headers of the message.
+ *
+ * This is not a complete implementation of BIMI.
+ *
+ * Copyright (c) 2023 Philippe Lieser
+ *
+ * This software is licensed under the terms of the MIT License.
+ *
+ * The above copyright and license notice shall be
+ * included in all copies or substantial portions of the Software.
+ */
+ 
+// options for ESLint
+/* global Components */
+/* global Logging */
+/* exported EXPORTED_SYMBOLS, BIMI */
+
+"use strict";
+
+var EXPORTED_SYMBOLS = [
+	"BIMI"
+];
+
+// @ts-ignore
+const Cu = Components.utils;
+
+Cu.import("resource://dkim_verifier/logging.jsm.js");
+Cu.import("resource://dkim_verifier/ARHParser.jsm.js");
+
+let BIMI = (function() {	
+	
+	const log = Logging.getLogger("BIMI");
+
+	// WSP as specified in Appendix B.1 of RFC 5234
+	const WSP_p = "[ \t]";
+	// obs-FWS as specified in Section 4.2 of RFC 5322
+	const obs_FWS_p = `(?:${WSP_p}+(?:\r\n${WSP_p}+)*)`;
+	// FWS as specified in Section 3.2.2 of RFC 5322
+	const FWS_p = `(?:(?:(?:${WSP_p}*\r\n)?${WSP_p}+)|${obs_FWS_p})`;
+
+	let that = {
+		/**
+		* Try to get the BIMI Indicator if available.
+		*
+		* @param {Map<string, string[]>} headers
+		* @param {Object[]} arhBIMI - Trusted ARHs containing a BIMI result.
```

Comment:
You should be able to specify the expected type here with `ARHResinfo` if you want, similar to master.

What you tried before (`ARHParser.ArhResInfo`) did not work because:
- The module system I use on master is understood by typescript, what is why I need to kind of import the type with `import("./arhParser.mjs.js")`. But like I wrote with the old system TypeScript assumes all to be in the same scope.
- The type name is case sensitive, and the casing of the type changed in master.

Reply:
Ok, I guess, I managed it :)